File: src/main/java/thkoeln/dungeon/planet/domain/CompassDirection.java
package thkoeln.dungeon.planet.domain;

public enum CompassDirection {
    north, east, south, west;

    public CompassDirection getOppositeDirection() {
        switch( this ) {
            case north: return south;
            case east: return west;
            case south: return north;
            case west: return east;
        }
        return null;
    }
}

File: src/main/java/thkoeln/dungeon/planet/domain/CompassDirection.kt
package thkoeln.dungeon.planet.domain

enum class CompassDirection {
    north, east, south, west;

    val oppositeDirection: CompassDirection?
        get() {
            return when (this) {
                north -> south
                east -> west
                south -> north
                west -> east
            }
            return null
        }
}